chrome/content/zotero/xpcom/data/feeds.js

```diff
@@ -68,6 +68,51 @@ Zotero.Feeds = new function() {
 		delete this._cache.urlByLibraryID[libraryID];
 		delete this._cache.libraryIDByURL[url];
 	}
+
+	this.init = function () {
+		return this.scheduleNextFeedCheck();
+	}
+	
+	this.restoreFromJSON = Zotero.Promise.coroutine(function* (json, merge=false) {
+		Zotero.debug("Restoring feeds from remote JSON");
+		Zotero.debug(json);
+		if (merge) {
+			let syncedFeeds = Zotero.SyncedSettings.get(Zotero.Libraries.userLibraryID, 'feeds');
+			for (let url in json) {
+				if (syncedFeeds[url]) {
+					syncedFeeds[url].name = json[url].name;
+					syncedFeeds[url].cleanupAfter = json[url].cleanupAfter;
+					syncedFeeds[url].refreshInterval = json[url].refreshInterval;
+					for (let guid in json[url].markedAsRead) {
+						syncedFeeds[url].markedAsRead[guid] = true;
+					}
+				} else {
+					syncedFeeds[url] = json[url];
+				}
+			}
+			json = syncedFeeds;
+		}
+		yield Zotero.SyncedSettings.set(Zotero.Libraries.userLibraryID, 'feeds', json);
+		//
+		let feeds = Zotero.Feeds.getAll();
+		for (let feed of feeds) {
+			if (json[feed.url]) {
+				Zotero.debug("Feed " + feed.url + " is being updated from remote JSON");
+				yield feed.updateFromJSON(json[feed.url]);
+				delete json[feed.url];
+			} else {
+				Zotero.debug("Feed " + feed.url + " does not exist in remote JSON. Deleting");
+				yield feed.eraseTx();
+			}
+		}
+		// Because existing json[feed.url] got deleted, `json` now only contains new feeds
+		for (let url in json) {
+			Zotero.debug("Feed " + url + " is being created from remote JSON");
+			let feed = new Zotero.Feed(json[url]);
+			yield feed.saveTx();
+			yield feed.updateFromJSON(json[url]);
```

Comment:
It is. From [these fields](https://github.com/zotero/zotero/pull/902/files#diff-d953027eb50324dcf90b1cd22cfe7eaaR340) `feed.updateFromJSON()` uses `markedAsRead` and the rest of the fields are used by the constructor. I anticipated `feed.updateFromJSON()` potentially needing other fields later, so I passed the full JSON, but we could only pass `markedAsRead` to reduce confusion.